MathLib/Point3dWithID.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * \file
+ * \date   2015-05-18
+ * \brief  Definition of the Point3d class.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef POINT3DWITHID_H_
+#define POINT3DWITHID_H_
+
+#include <limits>
+
+#include "Point3d.h"
+
+namespace MathLib
+{
+/**
+ * Class Point3dWithID is derived from class Point3d in
+ * order to extend the class Point3d with an id.
+ */
+class Point3dWithID: public Point3d {
+public:
+	/// Constructs a point with the coordinates x0, x1 and x2 and the provided
+	/// id.
+	/// @param x0 x coordinate of point
+	/// @param x1 y coordinate of point
+	/// @param x2 z coordinate of point
+	/// @param id the id of the object [default: max of std::size_t]
+	Point3dWithID(double x0, double x1, double x2,
+		std::size_t id = std::numeric_limits<std::size_t>::max())
+		: Point3d(std::array<double,3>({{x0, x1, x2}})), _id(id)
+	{}
+
+	/// Constructs a point using std::array<double,3> as coordinates and
+	/// the provided id.
+	/// @param coords coordinates of the point
+	/// @param id the id of the object [default: max of std::size_t]
+	Point3dWithID(std::array<double,3> & coords,
+		std::size_t id = std::numeric_limits<std::size_t>::max())
+		: Point3d(coords), _id(id)
+	{}
+
+	/// Constructs a point with the same coordinates as the given
+	/// Point3d pnt and the provided id.
+	/// @param pnt a MathLib::Point3d object containing the coordinates
+	/// @param id the id of the object [default: max of std::size_t]
+	Point3dWithID(MathLib::Point3d const& pnt,
```

Comment:
have you tested it? though i am not sure, since Point3d is a base class of Point3dWithID, compilers might use this constructor as a copy constructor.